### PR TITLE
Add pan, zoom, selection toggle to toolbar

### PR DESCRIPTION
**Pull request by VictoriaBeilsten-Edmands**
Button added to toggle between pan & wheel zoom, select to zoom and select region modes.

### CODE REVIEW

File: client/component/HeatmapPlot.tsx [hunk 1]
```diff
@@ -162,8 +181,14 @@ function HeatmapPlot(props: HeatmapPlotProps) {
             value: props.axesParameters.yValues?.data,
           } as AxisParams
         }
+        interactions={interactionsConfig}
       >
         <SelectionComponent
+          modifierKey={
+            mode === 'selection'
+              ? ([] as ModifierKey[])
+              : ('Shift' as ModifierKey)
```
Comment (PeterC-DLS) [inline client/component/HeatmapPlot.tsx:190]: Disable when not selected by toggle and remove modifier key

File: client/component/ImagePlot.tsx [hunk 2]
```diff
@@ -35,6 +44,16 @@ function ImagePlot(props: ImagePlotProps) {
   return (
     <>
       <Toolbar>
+        <ToggleGroup
```
Comment (PeterC-DLS) [inline client/component/ImagePlot.tsx:47]: Refactor to common toolbar items

File: client/component/utils.ts [hunk 3]
```diff
@@ -354,6 +361,21 @@ function isValidPositiveNumber(
   return [Number.isFinite(n) && n > 0 && n < upper, n];
 }
 
+function createInteractionsConfig(
+  mode: 'pan' | 'zoom' | 'selection'
+): DefaultInteractionsConfig {
+  return {
+    pan: mode === 'pan' ? ({} as PanProps) : false,
```
Comment (PeterC-DLS) [inline client/component/utils.ts:368]: Need to enable/disable all these according to the `mode`

File: client/component/LinePlot.tsx [hunk 4]
```diff
@@ -85,10 +89,24 @@ function LinePlot(props: LinePlotProps) {
...
   return (
     <>
       <Toolbar>
+        <ToggleGroup
+          role="radiogroup"
+          ariaLabel="mode"
+          value={mode}
+          onChange={setMode}
+        >
+          <ToggleGroup.Btn label="pan" value="pan" />
```
Comment (PeterC-DLS) [inline client/component/LinePlot.tsx:106]: The three categories are "pan & wheel zoom", "select to zoom" and "select region". The latter will be extended with different region types but start off with a rectangle.

**Review by PeterC-DLS — COMMENTED** [review]
Infrastructure is fine. Needs a little tweaking so make interactions more uniform so `Alt` does x-axis and `Shift` does y-axis.

File: client/component/SelectionComponent.tsx [hunk 5]
```diff
@@ -1,30 +1,44 @@
...
-          <SvgElement>
-            <SvgRect
-              coords={htmlSelection as Rect}
-              fill={isValid ? 'blue' : 'orangered'}
-              fillOpacity="0.3"
-            />
-          </SvgElement>
-        )}
-      </SelectionTool>
+      {disabled ? null : (
```
Comment (PeterC-DLS) [inline client/component/SelectionComponent.tsx:23]: SelectionTool has a `disabled` prop

**Review by PeterC-DLS — COMMENTED** [review]
Is it possible to use icons and tooltips for the zoom buttons?

**Comment by VictoriaBeilsten-Edmands** [comment]
> Is it possible to use icons and tooltips for the zoom buttons?

As in to change the cursor to different icons in the different modes and tooltips to explain the modifier key combinations?

**Comment by PeterC-DLS** [comment]
> > Is it possible to use icons and tooltips for the zoom buttons?
> 
> As in to change the cursor to different icons in the different modes and tooltips to explain the modifier key combinations?

No, use icons instead of text for the buttons. Can change the mouse pointer in another PR.

**Comment by VictoriaBeilsten-Edmands** [comment]
> 

Yes, there are react-icons [here](https://react-icons.github.io/react-icons) that we could use.

**Comment by PeterC-DLS** [comment]
TbZoomInArea and TbZoomPan seem suitable

File: client/component/ToggleGroup.tsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    position = 'left',
+  } = props;
+  const {
+    role,
+    value: selectedValue,
+    disabled: isGroupDisabled,
+    onChange,
+  } = useToggleGroupProps();
+
+  let btn_class: string;
```
Comment (PeterC-DLS) [inline client/component/ToggleGroup.tsx:52]: I think you could make your own button with the tooltip inside it and pass the text as a prop then you re-use `h5web/lib`'s `ToggleGroup`

**Review by PeterC-DLS — APPROVED** [review]

**Pull request merged**